js/stdlib/ts/ETH_like.ts

```diff
@@ -276,24 +276,39 @@ const getMinBlock = (logs: any[]) =>
       ? (x.logIndex < acc.logIndex ? x : acc)
       : (x.blockNumber.toString() < acc.blockNumber.toString() ? x : acc), logs[0]);
 
+const getMaxBlock = (logs: any[]) =>
+  logs.reduce((acc: Log, x: Log) =>
+    (x.blockNumber == acc.blockNumber)
+      ? (x.logIndex > acc.logIndex ? x : acc)
+      : (x.blockNumber.toString() > acc.blockNumber.toString() ? x : acc), logs[0]);
+
+
 class EventCache {
 
-  cache: any[] = []
+  cache: any[] = [];
+
+  currentBlock = 0;
 
   constructor() {
     this.cache = [];
   }
 
-  async query_(fromBlock: number, toBlock: number, topic: string, getLogs: () => Promise<any[]>) {
+  async query_(fromBlock: number, toBlock: number, topic: string, getLogs: (currentBlock: number) => Promise<any[]>) {
```

Comment:
I continue to not read this closely :)

Reply:
Should I read the ETH code closely now? Or are you still working on it?

Reply:
yeah, I think it's good now